client/src/components/single-dog/SingleDog.js

```diff
@@ -18,6 +19,13 @@ export default function SingleDog() {
   const [friends, setFriends] = useState([]);
   let location = useLocation();
   let { id } = useParams();
+  let history = useHistory();
+
+  function handleDeleteFriend(friend) {
+    removeItemFromLocalStorageById("friends", friend.id);
+    setFriends(getItemsFromLocalStorage("friends"));
```

Comment:
I'm a bit confused because you have `friends` in this page, where a `SingleDog` only is supposed to be rendered.  Why do you need that?